services/payments/src/simcore_service_payments/api/rpc/_payments.py

```diff
@@ -86,3 +94,57 @@ async def init_payment(
         payment_id=f"{payment_id}",
         payment_form_url=f"{submission_link}",
     )
+
+
+@router.expose()
+async def cancel_payment(
+    app: FastAPI,
+    *,
+    payment_id: PaymentID,
+    user_id: UserID,
+    wallet_id: WalletID,
+) -> None:
+
+    # validation
+    repo = PaymentsTransactionsRepo(db_engine=app.state.engine)
+    payment = await repo.get_payment_transaction(
+        payment_id=payment_id, user_id=user_id, wallet_id=wallet_id
+    )
+
+    if payment is None:
+        raise PaymentNotFoundError(payment_id=payment_id)
+
+    if payment.state.is_completed():
+        if payment.state == PaymentTransactionState.CANCELED:
+            # Avoids error if multiple cancel calls
+            return
```

Comment:
:+1: There will be a similar situation for auto recharge top up that we will need to handle (to not top up multiple times)